packages/fx-core/src/core/FxCore.ts

```diff
@@ -1471,6 +1474,20 @@ export class FxCore implements v3.ICore {
     return ok(Void);
   }
 
+  // a project version check
+  async projectVersionCheck(inputs: Inputs): Promise<Result<VersionCheckRes, FxError>> {
```

Comment:
What's the scope of this function? Is it just tell whether it's a V2 project (versionInfo.isSupported = compatible) or not (versionInfo.isSupported = unsupported)?

Reply:
In vs 17.5, no project is upgradable, versionInfo.isSupported could be compatible for V2 project and unsupported for V3 project.